content/en/docs/tasks/administer-cluster/migrating-from-dockershim/migrate-dockershim-dockerd.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "Migrate Docker Engine nodes from dockershim to cri-dockerd"
+weight: 9
+content_type: task 
+---
+
+{{% thirdparty-content %}}
+
+This page shows you how to migrate your Docker Engine nodes to use `cri-dockerd`
+instead of dockershim. Follow these steps if your clusters run Kubernetes 1.23
+or earlier and you want to continue using Docker as your container runtime after
+you upgrade to Kubernetes 1.24 and later.
+
+## Overview
+
+In Kubernetes 1.23 and earlier, Docker Engine used a component called the
+dockershim to interact with Kubernetes system components such as the kubelet. In
+Kubernetes 1.24, the dockershim is being removed. A third-party replacement,
+`cri-dockerd`, is available if you want to continue using Docker Engine in
+Kubernetes 1.24 and later.
+
+{{<note>}}
+If you already use `cri-dockerd`, you aren't affected by the dockershim removal.
+Before you begin, [Check whether your nodes use the dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/find-out-runtime-you-use/).
+{{</note>}}
+
+If you do need to migrate to `cri-dockerd` so that you can continue using Docker
+as your container runtime, you should do the following for each affected node:
+
+1.  Install `cri-dockerd`.
+1.  Cordon and drain the node.
+1.  Configure the kubelet to use `cri-dockerd`. 
+1.  Restart the kubelet.
+1.  Verify that the node is healthy.
+
+We recommend that you test the migration on non-critical nodes first.
+
+You should perform the following steps for each node that you want to migrate
+off dockershim.
+
+## {{% heading "prerequisites" %}}
+
+*   [Install and start the `cri-dockerd` service on each node](https://github.com/mirantis/cri-dockerd#build-and-install).
+
+## Cordon and drain the node
+
+1.  Cordon the node to stop new Pods scheduling on it:
+
+    ```shell
+    kubectl cordon <NODE_NAME>
+    ```
+    Replace `<NODE_NAME>` with the name of the node.
+
+1.  Drain the node to safely evict running Pods: 
+
+    ```shell
+    kubectl drain <NODE_NAME> \
+        --ignore-daemonsets
+    ```
+
+## Configure the kubelet to use cri-dockerd
+
+The following steps apply to clusters set up using the kubeadm tool. If you use
+a different tool, you should modify the kubelet using the configuration
+instructions for that tool.
+
+1.  Open `/var/lib/kubelet/kubeadm-flags.env` 
+1.  Modify the `--container-runtime-endpoint` flag to
+    `unix:///var/run/cri-dockerd.sock`.
+
+`kubeadm` stores the socket for each host as an annotation in the `Node` object
+for the host. To change the annotation:
+
+1.  Open the `admin.conf` file to edit the `Node` object:
+
+    ```shell
+    kubectl edit no <NODE_NAME>
+    ```
+    This command opens the file in a text editor.
```

Comment:
To me, these lines don't yet make sense.

Reply:
`kubectl edit no <NODE_NAME>` uses the `admin.conf` to connect the the api-server and edit the node object.

you can remove / edit the line:
> Open the `admin.conf` file to edit the `Node` object:

Reply:
Updated. PTAL and tell me if it's still off-looking: 

For each node, `kubeadm` stores the socket as an annotation in the `Node` object
for that node. To change the annotation:

1. Edit the `Node` object: 

    ```shell
    kubectl edit no <NODE_NAME>
    ```
    This command opens the `admin.conf` file in a text editor. This file is only
    available on control plane nodes.

Reply:
I haven't checked, but: I'm fairly sure that `admin.conf` is a configuration file for the `kubectl` command line tool.
If that's right then I'd expect the command to run is:
```shell
KUBECONFIG=/path/to/admin.conf kubectl edit node <nodename>
```

Reply:
and the thing that gets opened in a text editor is a YAML representation of that Node.

Reply:
I like that description, makes sense. I'll wait for @neolit123 to confirm and then change the content to reflect that :) thanks for the blazing fast reviews y'all

Reply:
What @sftim is saying is correct.